src/lib/pages/backoffice/Loan/LoanDetails/LoanOverbookedWarning/LoanOverbookedWarning.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withCancel } from '@api/utils';
+import { documentApi } from '@api/documents';
+import { Message } from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+import {
+  BackOfficeRoutes,
+} from '@routes/urls';
+
+const OVERBOOKED_STATES = {
+    OVERBOOKED: "overbooked",
+    NOT_OVERBOOKED : "not_overbooked",
+    ERROR: "error",
+}
+
+export default class LoanOverbookedWarning extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      overbooked: "false",
+    };
+  }
+
+  componentDidMount() {
+    this.fetchCirculation()
+  }
+
+  componentWillUnmount() {
+    this.cancellableCirculation && this.cancellableCirculation.cancel();
+  }
+
+  fetchCirculation = async () => {
+    const { documentPid } = this.props;
+    try {
+      this.cancellableCirculation = withCancel(
+        documentApi.get(documentPid)
+      );
+      const response = await this.cancellableCirculation.promise;
+      const circulation = response.data.metadata.circulation;
+      console.log(circulation.overbooked)
+      this.setState({ overbooked: circulation.overbooked ? OVERBOOKED_STATES.OVERBOOKED : OVERBOOKED_STATES.NOT_OVERBOOKED});
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({ overbooked: OVERBOOKED_STATES.ERROR });
+      }
+    }
+  };
+
+
+  render() {
+    const { overbooked } = this.state;
+    const { documentPid } = this.props;
+    return (
+      <>
+      {overbooked != OVERBOOKED_STATES.NOT_OVERBOOKED &&
+        <Message warning>
+          {overbooked != OVERBOOKED_STATES.ERROR
+            ? <>This literature is <Link to={BackOfficeRoutes.documentDetailsFor(documentPid)}><code>overbooked</code></Link>.</>
+            : <>Cannot verify if this literature is overbooked. Please refresh the page.</>
+          }
+        </Message>
```

Comment:
I don't like this if inside an if, but I don't see any other way of implementing the <Link>.
I tried using string templates (and passing the Link object to the template) but never managed to display it properly.

Reply:
I would have implemented differently to avoid the extra ifs.

Define a couple of constants on the top:
```
OVERBOOKED_MSG = (documentPid) => <>This literature is <Link to={BackOfficeRoutes.documentDetailsFor(documentPid)}>overbooked</Link></>
ERROR_MSG = "Cannot verify if this literature is overbooked. Please refresh the page."
```

Default state in the constructor:
```
state = {
    hasError: false,
    msg: ''
}
```

When fetching the document:
```
fetchCirculation = async () => {
... if overbooked ...
this.setState({hasError: false, msg: OVERBOOKED_MSG(documentPid)});
... if error ...
this.setState({hasError: true, msg: ERROR_MSG});
```

In the render method:
```
const { hasError, msg } = this.state;
const color = hasError ? 'red' : 'yellow';
return msg && (
  <Message color={color}>
      {msg}
  </Message>
)
```

How does it look like this?